contracts/src/Teleporter/TeleporterMessenger.sol

```diff
@@ -126,9 +124,7 @@ contract TeleporterMessenger is ITeleporterMessenger, ReentrancyGuards {
                 requiredGasLimit: messageInput.requiredGasLimit,
                 allowedRelayerAddresses: messageInput.allowedRelayerAddresses,
                 message: messageInput.message,
-                receipts: _getOutstandingReceiptsToSend(
-                    messageInput.destinationChainID
-                )
+                receipts: outstandingReceipts[messageInput.destinationChainID].getOutstandingReceiptsToSend()
```

Comment:
Is this Solidity syntactic sugar for `getOutstandingReceiptsToSend(outstandingReceipts[messageInput.destinationChainID])`? The signature of this function is
```
function getOutstandingReceiptsToSend(
        TeleporterMessageReceiptQueue storage queue
)
```
so I'm trying to understand how dot-notation is used here.

Reply:
This is because we have `using ReceiptQueue for ReceiptQueue.TeleporterMessageReceiptQueue;` at beginning of the TeleporterMessenger contract.

Reply:
Gotcha. From the documentation, the way it works here is that we're taking the functions defined in `ReceiptQueue.sol` and adding them as operators for the type `ReceiptQueue.TeleporterMessageReceiptQueue`. So when we call using dot-notation here, this implicitly inserts the object as the first argument to the function call in `ReceiptQueue.sol`. 

I'm generally not a fan of this kind of aliasing/implicit function call conversion/operator overloading. However, it seems to be standard Solidity practice so I'm probably just being a curmudgeon :)